gordon/resources/s3.py

```diff
@@ -265,13 +265,13 @@ def __init__(self, *args, **kwargs):
 
     def get_bucket_arn(self):
         bucket = self.settings.get('bucket')
-        if bucket.startswith('arn:aws:'):
+        if isinstance(bucket, six.string_types) and bucket.startswith('arn:aws:'):
             return bucket
         return troposphere.Join("", ["arn:aws:s3:::", bucket])
```

Comment:
@bsiemon you should probably checkout the troposphere repo. It has some examples of using `troposphere.Join` for exactly that purpose: https://github.com/cloudtools/troposphere/blob/master/examples/Route53_RoundRobin.py#L29